### PR TITLE
Add internal panel types and misc cleanup

### PR DESCRIPTION
**Pull request by zapp-brannigan-dota**
Internal panel types are panel types that are not defined in any layout file. I attempted to load them in Panorama and parsed cl_panorama_script_help_2. I think that's most of them but I don't know how to check.

### CODE REVIEW

File: declarations/server/dota-api.d.ts [hunk 1]
```diff
@@ -1,3 +1,8 @@
+/// <reference path="dota-enums.d.ts" />
+/// <reference path="dota-gameevents.d.ts" />
+/// <reference path="dota-modifier-properties.d.ts" />
+/// <reference path="dota-std.d.ts" />
+
```
Comment (Perryvw) [inline declarations/server/dota-api.d.ts]: These shouldn't be in here.

File: declarations/panorama/dota.d.ts [hunk 2]
```diff
@@ -2303,7 +2306,8 @@ interface CPanoramaScript_VRUtils {
 
 interface DollarStatic {
     (selector: string) : Panel;
-    CreatePanel(type: string, root: Panel, name: string): Panel;
+    CreatePanel(type: string, root: Panel, id: string): Panel;
+    CreatePanel(type: string, root: Panel, id: string): PanelBase;
```
Comment (Perryvw) [inline declarations/panorama/dota.d.ts:2307]: PanelBase is not a thing? remove this line.

File: declarations/panorama/dota.d.ts [hunk 3]
```diff
@@ -11,6 +11,9 @@
     https://github.com/ModDota/API.
 */
 
+/// <reference path="dota_enums.d.ts" />
+/// <reference path="dota_panels.d.ts" />
+
```
Comment (Perryvw) [inline declarations/panorama/dota.d.ts]: These shouldn't be in here

**Review by Perryvw — CHANGES_REQUESTED** [review]

File: declarations/panorama/dota.d.ts (hunk 2 above)
Comment (zapp-brannigan-dota) [reply declarations/panorama/dota.d.ts:2307]: PanelBase is a parent to Panel, SliderPanel and CircularProgressBar.
SliderPanel and CircularProgressBar are missing all but four Panel properties, so I moved those four into a separate interface.

File: declarations/server/dota-api.d.ts (hunk 1 above)
Comment (zapp-brannigan-dota) [reply declarations/server/dota-api.d.ts]: They make it possible to reference dota-api.d.ts with something like
```/// <reference path="../../../../declarations/server/dota-api.d.ts" />```

I have a folder where I download the latest declarations into and reference them like this. Others might want to do the same so I think they should stay.

Comment (Perryvw) [reply declarations/server/dota-api.d.ts]: Should be in your code, not in declarations.

File: declarations/panorama/dota_panels.d.ts [hunk 4]
```diff
@@ -130,84 +137,62 @@ interface Panel {
...
-    SetPanelEvent(event: PanelEvent, handler: () => void): void;
-
-    RunScriptInPanelContext(script: string): void;
 }
 
-interface VCSSStyleDeclaration extends CSSStyleDeclaration {
-    /** 
-     * Controls blending mode for the panel.  See CSS mix-blend-mode docs on web, except normal for us is with alpha blending.
-     * 
+interface VCSSStyleDeclaration {
```
Comment (Perryvw) [inline declarations/panorama/dota_panels.d.ts:148]: Why is the extension gone? Are its fields not nullable?

File: declarations/server/dota-api.d.ts (hunk 1 above)
Comment (zapp-brannigan-dota) [reply declarations/server/dota-api.d.ts]: Ah ok.

File: declarations/panorama/dota_panels.d.ts (hunk 4 above)
Comment (zapp-brannigan-dota) [reply declarations/panorama/dota_panels.d.ts:148]: To improve autocomplete results. All possible values are included.

**Review by Perryvw — APPROVED** [review]

**Pull request merged**